### PR TITLE
[dev] core: spclient: parse lyrics

### PR DESCRIPTION
**Pull request by gdesmott**
Safe users from doing all the parsing themselves.

### CODE REVIEW

**Comment by roderickvd** [comment]
Thanks for this! This has been missing from `metadata`. Would you be so kind to move the struct and parsing there? The idea is that `SpClient` is a fairly low-level interface and like the other responses, they are parsed in `metadata`.

**Comment by roderickvd** [comment]
It's been a while since I last worked on `metadata` -- I can't remember if there are any structs there that do not parse a protobuf. I would not mind if you did not implement the `Metadata` trait but do a custom implementation of `request` and/or `parse`. If you see any way to make it more generic, feel free to do so.

**Comment by gdesmott** [comment]
Something like this then?

File: metadata/src/lyrics.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+use bytes::Bytes;
+
+use librespot_core::{Error, FileId, Session, SpotifyId};
+
+impl LyricsReply {
```
Comment (roderickvd) [inline metadata/src/lyrics.rs:5]: Maybe just `Lyrics`?

File: metadata/src/lyrics.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    pub async fn get(session: &Session, id: &SpotifyId) -> Result<Self, Error> {
+        let spclient = session.spclient();
+        let lyrics = spclient.get_lyrics(id).await?;
+        LyricsReply::try_from(&lyrics)
```
Comment (roderickvd) [inline metadata/src/lyrics.rs:9]: You could return `Self` here.

File: metadata/src/lyrics.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+}
+
+impl TryFrom<&Bytes> for LyricsReply {
+    type Error = Error;
+
+    fn try_from(lyrics: &Bytes) -> Result<Self, Self::Error> {
+        let parsed = serde_json::from_slice(lyrics)?;
+
+        Ok(parsed)
```
Comment (roderickvd) [inline metadata/src/lyrics.rs:29]: Probably you could return `serde_json::from_slice(lyrics)` immediately? Maybe with `map_err(Into::into)`?

**Review by roderickvd — COMMENTED** [review]
Yeah something like that! A few comments.

File: metadata/src/lyrics.rs (hunk 1 above)
Comment (gdesmott) [reply metadata/src/lyrics.rs:5]: Agreed, but then I need to change the name of the other inner struct. `LyricsInner` ? `LyricsDetails`?

Comment (roderickvd) [reply metadata/src/lyrics.rs:5]: Inner sounds good to me.

**Comment by gdesmott** [comment]
@roderickvd I fixed your comments

**Comment by roderickvd** [comment]
Looks good to me. If you could add a changelog entry, ready to merge.
Feel free to fix that new clippy lint along the way to fix the CI.

**Comment by gdesmott** [comment]
done

**Comment by gdesmott** [comment]
All clippy warnings are now fixed.

**Comment by roderickvd** [comment]
Great ❤️

**Pull request merged**